### PR TITLE
Andrewrutter/kkb 67 signup page frontend update

### PR DESCRIPTION
**Pull request by andlrutt**
This PR updates the event description page and by extension the event signup component. It also incorporates the group functionality, as well as if no signups are required for an event.  
A few things to note:

- The backend functionality is dependent on outstanding PRs, so it hasn't been properly tested just yet.
- Added a new color in the theme for the donate button
- The donate button currently has no functionality

EDIT: I have added donate button functionality and tested it. it works!

Default appearance
image

when groupSignUp = true
image

when maxVolunteers = 0
image

### CODE REVIEW

**Review by kemalfidan — CHANGES_REQUESTED** [review]
Just merged the backend updates for this into `develop` so you can hook up the frontend to it now. I noticed the API request might need to be updated and there might be some conflict since I already added the `groupSignUp` from to our types and schema. You have most of what you have seems done so just rerequest the review when you make the changes and I'll take a better look.

File: src/components/EventSignUp/index.tsx [hunk 1]
```diff
@@ -60,173 +62,181 @@ const EventSignUp: React.FC<Props> = ({ id }) => {
...
+                    type="number"
+                    name="groupCount"
+                    onChange={handleGroupCountChange}
+                    placeholder="Group Count (Optional)"
+                    className={styles.otherInput}
+                    id="groupCountField"
+                />
+            );
+        }
+    };
```
Comment (kemalfidan) [inline src/components/EventSignUp/index.tsx:89]: Here you can do ```{ groupSignUp && <input ... /> }``` for conditional rendering. Not worth changing now but just wanted to let you know of an easier way.

**Review by kemalfidan — APPROVED** [review]
👍

File: src/components/EventSignUp/index.tsx (hunk 1 above)
Comment (andlrutt) [reply src/components/EventSignUp/index.tsx:89]: Ah, thanks for the tip!

**Review by rossketron — APPROVED** [review]
Looks great!

**Pull request merged**